### PR TITLE
Replace waitAndAssert in config_test.go with poll.WaitOn

### PR DESCRIPTION
**Pull request by yongtang**
This fix replaces waitAndAssert in config_test.go with poll.WaitOn
so that the testing is consistent with all other tests in integration.

Also, config_test.go uses to wait and sleep for 2 * (1 minutes) to get the task
info. This fix combined those two sleep and wait for 1 mins. Think 1 min
is enough for config test.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
oh! in case you were thinking on moving the `SwarmLockUnlock` tests to `integration/`. I had this written for the NodeStatus; https://github.com/moby/moby/compare/master...thaJeztah:poll_node 😅

**Comment by olljanat** [comment]
Derek add label: rebuild/*

File: integration/config/config_test.go [hunk 1]
```diff
@@ -267,19 +268,32 @@ func TestTemplatedConfig(t *testing.T) {
...
+				return poll.Success()
+			}
+			return poll.Continue("task still waiting")
+		default:
+			return poll.Continue("task still waiting")
 		}
-		return task.NodeID != "" && task.Status.ContainerStatus != nil && task.Status.ContainerStatus.ContainerID != ""
-	})
+	}
+	poll.WaitOn(t, getServiceTask, swarm.ServicePoll, poll.WithTimeout(1*time.Minute))
```
Comment (thaJeztah) [inline integration/config/config_test.go]: I wonder if we should split this into two steps;

- check if the service reconciled (check if the desired number of tasks is running); that may be more generic / reusable
- once that's the case, do the `inspect` on its task

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Left some ideas; actually part of that code written while reviewing earlier today, so I pushed it as a branch; https://github.com/moby/moby/pull/38558 😅

**Comment by yongtang** [comment]
Thanks @thaJeztah for the review. I have updated the PR and used the changes you made to `swarm.GetRunningTasks`. Please take a look

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Pull request merged**